### ISSUE TITLE
Unable to compile

### ISSUE THREAD

**Issue by vesatoivonen** [keenerd/gz-sort#5]
```
$ make
cc -O3 -std=gnu99 -Wall -Werror -pedantic -Wextra -pthread    -c -o gz-sort.o gz-sort.c
gz-sort.c: In function ‘nway_chop_and_presort’:
gz-sort.c:409:5: error: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Werror=unused-result]
     asprintf(&report, "%s line count: %ld\n%s %s", misc->label, out.line_counter, misc->label, "chop/presort");
     ^
gz-sort.c: In function ‘first_pass’:
gz-sort.c:433:5: error: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Werror=unused-result]
     asprintf(&report, "%s line count: %ld\n%s %s", misc->label, in1.line_counter, misc->label, label2);
     ^
gz-sort.c: In function ‘middle_passes’:
gz-sort.c:558:9: error: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Werror=unused-result]
         asprintf(&report, "%s merge %ld", misc->label, average);
         ^
gz-sort.c: In function ‘nway_merge_pass’:
gz-sort.c:679:5: error: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Werror=unused-result]
     asprintf(&report, "%i-way merge", misc->nway);
     ^
gz-sort.c: In function ‘main’:
gz-sort.c:791:9: error: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Werror=unused-result]
         asprintf(&temp_path, "%s.temp", output_path);
         ^
gz-sort.c:810:9: error: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Werror=unused-result]
         asprintf(&(nway_table[i].label), "T%i", i+1);
         ^
gz-sort.c:811:9: error: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Werror=unused-result]
         asprintf(&(nway_table[i].in_path), "%s.T%i.temp", output_path, i+1);
         ^
gz-sort.c:812:9: error: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Werror=unused-result]
         asprintf(&(nway_table[i].out_path), "%s.T%i.gz",  output_path, i+1);
         ^
cc1: all warnings being treated as errors
<builtin>: recipe for target 'gz-sort.o' failed
make: *** [gz-sort.o] Error 1
```

**Comment by keenerd** [comment]
You are missing every important detail like OS version and compiler version.

**Comment by vesatoivonen** [comment]
Ubuntu 16.04.2 LTS
```
$ cc --version
cc (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609

```

**Comment by unhammer** [comment]
Also seeing this on xubuntu 17.04 with cc (Ubuntu 6.3.0-12ubuntu2) 6.3.0 20170406

**Comment by DtpEJsaYXDU4GDH8dE4MyI9VrieF0UZpPZ0K76K** [comment]
Still doesn't compile due to -Werror in Makefile

$ make
cc -O3 -std=gnu99 -Wall -Werror -pedantic -Wextra -pthread    -c -o gz-sort.o gz-sort.c
gz-sort.c: In function ‘main’:
gz-sort.c:761:17: error: this statement may fall through [-Werror=implicit-fallthrough=]
                 show_help();
                 ^~~~~~~~~~~
gz-sort.c:762:13: note: here
             default:
             ^~~~~~~
cc1: all warnings being treated as errors
<builtin>: recipe for target 'gz-sort.o' failed
make: *** [gz-sort.o] Error 1

**Comment by keenerd** [comment]
Those have been fixed in commits bae88e7 and 8f4b257.  Are there any other errors you are getting now?

**Comment by vesatoivonen** [comment]
I'm able to compile without errors now, thanks.
